client/js/preload/goldstoneRouter.js

```diff
@@ -37,6 +37,7 @@ var LauncherView = Backbone.View.extend({
 var GoldstoneRouter = Backbone.Router.extend({
     routes: {
         "api_perf/report": "apiPerfReport",
+        "apps/opentrail": "openTrail",
```

Comment:
I'm not sure I understand the requirements yet. The goldstoneRouter routes can be seen as 'vanity license plates'. They are arbitrary and easily changed. They are just loading client views that are indifferent to the underlying backend api calls. In terms of a dynamic solution, I will have to think about this, and am open to suggestions. 

fyi: url route /yourapp/ will be picked up by django.  url route /#yourapp/ will be picked up by gsRouter.
So i could use the "url_root" key to dynamically add a route for that.
Thinking...